examples/loader-rust-to-json/src/index.md

```diff
@@ -0,0 +1,232 @@
+                0 => Suit::Clubs,
+                1 => Suit::Diamonds,
+                2 => Suit::Hearts,
+                3 => Suit::Spades,
+                _ => unreachable!(),
+            };
+            let card = Card { rank, suit };
+            if cards.iter().any(|&c| c == card) { continue };
+            cards.push(card);
+        }
+        Self(cards)
+    }
+
+    fn categorize(&self) -> HandCategory {
+        let rank_counts = self.0.iter().fold(HashMap::new(), |mut acc, card| {
+            *acc.entry(card.rank).or_insert(0) += 1;
+            acc
+        });
+        let suit_counts = self.0.iter().fold(HashMap::new(), |mut acc, card| {
+            *acc.entry(card.suit).or_insert(0) += 1;
+            acc
+        });
+        let is_flush = suit_counts.len() == 1;
+        let is_straight = if self.0.iter().any(|card| card.rank == 1) {
+            // Handle aces
+            let min_rank = self.0.iter().map(|card| card.rank).filter(|&rank| rank != 1).min().unwrap();
+            let max_rank = self.0.iter().map(|card| card.rank).filter(|&rank| rank != 1).max().unwrap();
+            (min_rank == 2 && max_rank == 5) || (min_rank == 10 && max_rank == 13)
+        } else {
+            let min_rank = self.0.iter().map(|card| card.rank).min().unwrap();
+            let max_rank = self.0.iter().map(|card| card.rank).max().unwrap();
+            (max_rank - min_rank) as usize == self.0.len() - 1
+        };
+
+        if is_flush && is_straight {
+            HandCategory::StraightFlush
+        } else if rank_counts.values().any(|&count| count == 4) {
+            HandCategory::FourOfAKind
+        } else if rank_counts.values().any(|&count| count == 3)
+            && rank_counts.values().any(|&count| count == 2)
+        {
+            HandCategory::FullHouse
+        } else if is_flush {
+            HandCategory::Flush
+        } else if is_straight {
+            HandCategory::Straight
+        } else if rank_counts.values().any(|&count| count == 3) {
+            HandCategory::ThreeOfAKind
+        } else if rank_counts.values().filter(|&&count| count == 2).count() == 2 {
+            HandCategory::TwoPair
+        } else if rank_counts.values().any(|&count| count == 2) {
+            HandCategory::OnePair
+        } else {
+            HandCategory::HighCard
+        }
+    }
+}
+```
+
+<div class="note">
+
+To run this data loader, you’ll need [rust-script](https://rust-script.org/) installed.
+
+</div>
+
+The above data loader lives in `data/poker.json.rs`, so we can load the data as `data/poker.json` using `FileAttachment`.
+
+```js echo
+const hands = FileAttachment("data/poker.json").json();
+```
+
+<div style="max-width: 300px;">
+
+```js
+Inputs.table(hands.summary)
+```
+
+</div>
+
+Taking advantage of Rust’s performance and simple parallelism, the data loader above was able to summarize ${hands.meta.count / 1e6} million hands in ${hands.meta.duration_s.toFixed(1)} seconds.
```

Comment:
This shows as “the data loader above was able to summarize 10 million hands in 1.3 seconds” and might show as “1 seconds” for someone with a slightly faster machine. You could use milliseconds to avoid having to do a singular “1 second”.

Reply:
Good call!